### ISSUE TITLE
Unhandled exceptions in .Net6 WPF app.

### ISSUE THREAD

**Issue by gh615** [hardkoded/puppeteer-sharp#2109]
### Description

About unhandled exceptions in .Net6.
This exception doesn't happen every time, but it happens quite often when I try the code below.

### Complete minimal example reproducing the issue

1. Create a .Net6 WPF project.
1. Install Puppeteer Sharp with nuget.
1. Write the following code in App.xaml.cs.
   ```C#
   public partial class App : Application
   {
       public App() => SetupExceptionHandling();
   
       protected async override void OnStartup(StartupEventArgs e)
       {
           base.OnStartup(e);
           _ = Task.Run(async () =>
           {
               try
               {
                   using (var browser = 
                       await Puppeteer.LaunchAsync(new LaunchOptions
                       {
                           Headless = true,
                           Args = new string[] {
                               "--incognito",
                           },
                           ExecutablePath = @"path\to\chrome.exe",
                       })
                   )
                   using (var page = await browser.NewPageAsync())
                   {
                       await page.GoToAsync("https://google.com");
                       //var useragent = await page.EvaluateExpressionAsync<string>("navigator.userAgent");
                       await Task.Delay(3000);
                   }
                   System.Diagnostics.Debug.Print("End");
               }
               catch (Exception ex)
               {
   				System.Diagnostics.Debug.Print(ex.ToString());
               }
           });
           MessageBox.Show($"Wait until \"End\" is output to the Immediate Window, then click OK.", "test");
       }
   
       private void SetupExceptionHandling()
       {
           AppDomain.CurrentDomain.UnhandledException += (s, e) =>
           {
               if (!e.IsTerminating)
                   LogUnhandledException((Exception)e.ExceptionObject, "AppDomain.CurrentDomain.UnhandledException");
           };
   
           DispatcherUnhandledException += (s, e) =>
           {
               LogUnhandledException(e.Exception, "Application.Current.DispatcherUnhandledException");
               e.Handled = true;
           };
   
           TaskScheduler.UnobservedTaskException += (s, e) =>
           {
               LogUnhandledException(e.Exception, "TaskScheduler.UnobservedTaskException");
               e.SetObserved();
           };
       }
   
       private void LogUnhandledException(Exception exception, string source)
       {
           MessageBox.Show($"{source}: {exception.Message}\r\n{exception}", "");
           Environment.Exit(-1);
       }
   }
   ```
1. Build and run the project.
1. The exception is signaled by the LogUnhandledException method before or after closing the MessageBox. The contents of the exception are as follows.
```
TaskScheduler.UnobservedTaskException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property.
As a result, the unobserved exception was rethrown by the finalizer thread. (Response body is unavailable for redirect responses)
System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property.
As a result, the unobserved exception was rethrown by the finalizer thread. (Response body is unavailable for redirect responses)
 ---> PuppeteerSharp.PuppeteerException: Response body is unavailable for redirect responses
   --- End of inner exception stack trace ---
```

I've tried the same code in a .Net Framework 4.8 WPF project and it didn't happen no matter how many times I tried.
This seems to only occur in .Net 6.
In `OnStartup`, you can still get an error without running `MessageBox.Show`, but with `MessageBox.Show`, the error is more likely to occur.

My code doesn't catch any try-catch, so I think the exception is being thrown in the thread that PuppeteerSharp starts.

By the way, changing `await page.GoToAsync("https://google.com");` to `var useragent = await page.EvaluateExpressionAsync<string>("navigator.userAgent");` changes the content of the exception.

```
TaskScheduler.UnobservedTaskException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property.
As a result, the unobserved exception was rethrown by the finalizer thread. (The semaphore has been disposed.)
System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property.
As a result, the unobserved exception was rethrown by the finalizer thread. (The semaphore has been disposed.)
 ---> System.ObjectDisposedException: The semaphore has been disposed.
   at System.Threading.SemaphoreSlim.CheckDispose()
   at System.Threading.SemaphoreSlim.WaitAsync(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at PuppeteerSharp.Helpers.TaskQueue.DisposeAsyncCore() in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\Helpers\TaskQueue.cs:line 74
   at PuppeteerSharp.Helpers.TaskQueue.DisposeAsync() in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\Helpers\TaskQueue.cs:line 26
   --- End of inner exception stack trace ---
```

Is it possible to handle this exception? or is it safe to ignore it?

### Expected behavior:

Exceptions are Handled correctly.

### Actual behavior:

Unhandled Exception occurred.

### Versions

* Which version of PuppeteerSharp are you using? 
PuppeteerSharp 9.0.2
* Which .NET runtime and version are you targeting? E.g. .NET framework 4.6.1 or .NET Core 2.0.
.Net6 WPF

**Comment by gh615** [comment]
Hi, kblok!

Thank you for you reply.
I had already seen that issue but didn't think it was the same.

#1870 seems to occur in .NET framework 4.8, but the above issue does not occur in .NET framework 4.8.
Happened in .Net 6.
Are they really the same?

If so, the issue has already been closed, but is it a specification?

I'm afraid I don't have time to look deeply into the code right now, but I would appreciate it if you could tell me.
Thank you.

**Comment by kblok** [comment]
Hey, @gh615, the conclusion there was to ignore `UnobservedTaskException`. It was super hard to get rid of those, and no one was able to find a good solution for that.

**Comment by gh615** [comment]
Thank you for your kindness.
I understand.
I also ignore `UnobservedTaskException` raised by PuppeteerSharp.
Thank you!

**Comment by amaitland** [comment]
> System.ObjectDisposedException: The semaphore has been disposed.
>    at System.Threading.SemaphoreSlim.CheckDispose()
>    at System.Threading.SemaphoreSlim.WaitAsync(Int32 millisecondsTimeout, CancellationToken cancellationToken)
>    at PuppeteerSharp.Helpers.TaskQueue.DisposeAsyncCore() in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\Helpers\TaskQueue.cs:line 74

This particular call stack should be resolved by #2095

**Comment by gh615** [comment]
Thanks for the info!
